hera_pspec/uvwindow.py

```diff
@@ -0,0 +1,946 @@
+from pyuvdata import utils as uvutils
+import uvtools.dspec as dspec
+import h5py
+import warnings
+import numpy as np
+import sys, os, time
+from scipy.interpolate import interp2d
+from pyuvdata import UVBeam, UVData 
+
+from . import conversions, noise, version, pspecbeam, grouping, utils, uvpspec_utils as uvputils
+
+class UVWindow(object):
+    """
+    An object for storing window functions copmuted without the delay approximation.
+    """
+
+    def __init__(self, ftbeam='', uvdata='',taper='none', 
```

Comment:
Yes I'm currently working on this (future PR!). For now I have opted for adding a function to `UVPSpec` which computes the exact window functions of the object (something along the lines of `uvp.get_exact_window_functions`) and includes the `error_weights` if given. Since the `uvp.stats_array` retains the baseline-pair information even after averaging, we avoid the caveat of missing information. And a user warning could be sufficient to discourage someone from computing the exact window functions of an averaged `UVPSpec` object without using weights. 
I'd be happy to talk more about how to do all of this properly!

Reply:
I think I'm going to remove the option to initialise the `UVWindow` object with a `UVData` object because I am not sure how useful it is in the end. What is really useful is the interface with `UVPSpec` indeed